include/boost/geometry/algorithms/detail/is_valid/has_valid_self_turns.hpp

```diff
@@ -87,11 +87,13 @@ class has_valid_self_turns
                 is_acceptable_turn<Geometry>
             > interrupt_policy;
 
+        // Calculate self-turns, skipping adjacent segments
         detail::self_get_turn_points::self_turns<false, turn_policy>(geometry,
                                           strategy,
                                           robust_policy,
                                           turns,
-                                          interrupt_policy);
+                                          interrupt_policy,
+                                          0, true);
```

Comment:
What does this change do? Does this change validity check WRT how it was in the past or is it a change needed because some other change was done in `self_turns`?

Reply:
According to this comment: 

    // In some cases (dissolve, has_self_intersections)
    // neighbouring segments should be checked
    // (for example to detect spikes properly)

(https://github.com/boostorg/geometry/blob/develop/include/boost/geometry/algorithms/detail/overlay/get_turns.hpp#L393)

Though in `is_valid` spikes are detected separately here:

https://github.com/boostorg/geometry/blob/develop/include/boost/geometry/algorithms/detail/is_valid/ring.hpp#L210

May these adjacent turns be important for some other reason?

Reply:
Thanks, I will verify it.

Reply:
It is indeed wrong, I will revert it. Strangely enough the tests in `is_valid` did not catch it, so I will add a testcase there too.

It causes some false positives in my branch so I have to solve that in another way.

Will update this PR today

Reply:
Hmm, that was in my derived branch. In the original branch it is all OK, and also well tested.

Indeed, spikes are detected separately and therefore adjacent turns to detect turns in validity checking. So I think we can apply it as is now.

Agree?

Reply:
Ok, if there are problems in the future we can revert it later.

Reply:
Thank you, and a bug-label is OK